### PR TITLE
Update tests to support more types of SwiftPM build messages

### PR DESCRIPTION
**Pull request by hartbit**
This will allow merging the SwiftPM build animation without failing those tests: https://github.com/apple/swift-package-manager/pull/2050

### CODE REVIEW

**Comment by hartbit** [comment]
@aciidb0mb3r what do you think?

File: test-codecov-package/test-codecov-package.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // RUN: rm -rf %t.dir
 // RUN: mkdir -p %t.dir
 // RUN: cp -r %S/foo %t.dir/
-// RUN: %{swift-test} --package-path %t.dir/foo --enable-code-coverage -v 2>&1 | tee %t.build-log
+// RUN: %{swift} test --package-path %t.dir/foo --enable-code-coverage -v 2>&1 | tee %t.build-log
```
Comment (aciidb0mb3r) [inline test-codecov-package/test-codecov-package.txt]: why this change?

**Comment by aciidb0mb3r** [comment]
One minor comment but looks good, thanks!

File: test-codecov-package/test-codecov-package.txt (hunk 1 above)
Comment (hartbit) [reply test-codecov-package/test-codecov-package.txt]: This line is a revert from the previous PR: this specific test can't work correctly against a custom SwiftPM binary because it expects to find code coverage information in the same folder. It basically only works with a toolchain.

Comment (aciidb0mb3r) [reply test-codecov-package/test-codecov-package.txt]: Cool, feel free to test and merge this!

**Comment by hartbit** [comment]
@aciidb0mb3r I've tested this PR in https://github.com/apple/swift/pull/23417. Can you merge it for me? I don't have merge rights on this repo.
